### PR TITLE
Bugfixes in Invoice Module

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
Fix: #1974 

Bugs are elaborated in the below document with the module index as the Invoice.
https://docs.google.com/spreadsheets/d/1CB139UvzvUtB1JOHYP0nmEqjxdoHQPGWLGjD_niH9pw/edit#gid=1379931937

### CODE REVIEW

File: app/Http/Controllers/Order/InvoiceController.php [hunk 1]
```diff
@@ -208,10 +208,21 @@ public function getInvoices(Request $request)
...
+                           } elseif ($keyword == 'Unpaid') {
+                               $sql = 'status like ?';
+                               $sql2 = 'pending';
+                               $query->whereRaw($sql, ["%{$sql2}%"]);
+                           } elseif ($keyword == 'Partiallypaid' || $keyword == 'Partially') {
+                               $sql = 'status like ?';
+                               $sql2 = 'partially paid';
+                               $query->whereRaw($sql, ["%{$sql2}%"]);
+                           }
+                       })
```
Comment (ajaz-ul-haq) [inline app/Http/Controllers/Order/InvoiceController.php:225]: need to refactor whole block of code, 

Can't we use some thing like  below : 
                     ->filterColumn('status', function ($query, $keyword) {
                           $query->where('status','LIKE', '%'.$this->fetchKeyWordForQuery($keyword).'%' );
                       })
and 
    private function fetchKeyWordForQuery($keyword = 'Paid'): string
    {
        return match ($keyword){
            'Paid' => 'success',
            'Unpaid' => 'pending',
            'Partiallypaid' || 'Partially' => 'partially paid'
        };
    }

File: app/Http/Controllers/Order/InvoiceController.php [hunk 2]
```diff
@@ -208,26 +208,23 @@ public function getInvoices(Request $request)
...
 
+     private function fetchKeyWordForQuery($keyword = 'Paid'): string
+        {
+        return match ($keyword){
+        'Paid','paid' => 'success',
+        'Unpaid','unpaid' => 'pending',
+        'Partiallypaid','Partially','partiallypaid' => 'partially paid'
+        };
+        }
+
```
Comment (ajaz-ul-haq) [inline app/Http/Controllers/Order/InvoiceController.php:227]: please give proper indentation

Comment (sowmiyasubramaniyan) [reply app/Http/Controllers/Order/InvoiceController.php]: Done

**Review by ajaz-ul-haq — APPROVED** [review]

**Review by priyadarshini2101 — CHANGES_REQUESTED** [review]
- [ ] The search textbox is not working with the status Keyword “paid”.---- fixed but, not searching for the other words and there are UI changes in pdf alignment. Shown to sowmya. Once fixed will retest the pr.

**Review by priyadarshini2101 — DISMISSED** [review]
Tested the pr with server instance.

- [ ]  CIN, GSTIN, and Phone no. values are overlapping while generating pdf.

- [ ] Generate pdf alignment is not aligned properly.

feb7af2

- [ ] The search textbox is not working with the status Keyword “paid”.

feb7pd

**Review by SarfrazS — CHANGES_REQUESTED** [review]
- [ ]  The search textbox is not working with the status Keyword “paid”.  Not working in client panel.
image

**Review by SarfrazS — DISMISSED** [review]
Pointed issue fixed.
image

**Pull request merged**